collect/cnrfc/cnrfc.py

```diff
@@ -318,6 +318,65 @@ def get_deterministic_forecast_watershed(watershed, date_string, acre_feet=False
                                  'downloaded': dt.datetime.now().strftime('%Y-%m-%d %H:%M')}}
 
 
+def get_deterministic_forecast_watershed_IO(watershed, date_string, path):#, cnrfc_id=None):
```

Comment:
In general, the structures of these two functions are very similar after the creation of the URL, which is great! That lends itself well to consolidation. Perhaps something like:

```python
def the_eventual_function_name(watershed, date_string, forecast_type, duration=None, filename=None):
    """
    fill out docstring with description of function

    Arguments:
        watershed (str): the forecast group identifier
        date_string (str): the forecast issuance date as a YYYYMMDDHH formatted string
        forecast_type (str): one of deterministic or ensemble
        duration (str): forecast data timestep (hourly or daily)
        filename (str): file name including file path/s3 key
    Returns:
        filename (str): the destination forecast file name
    """
    # store original date_string
    _date_string = date_string

    # forecast datestamp prefix
    date_string = _default_date_string(date_string)
    
   # define the URL
    if forecast_type == 'deterministic':
         # use the watershed and updated date_string to define the URL
    elif forecast_type == 'ensemble':
        # use the watershed, updated date_string, and duration to define the URL

    url = 'https://www.cnrfc.noaa.gov/csv/' + the part that's different from above
    
    # extract the CSV data from the zip object
    # use similar structure here where you check if it's a valid url and handle bad zipfiles
    
    # define a default path if no filename provided
    
    # write the csvdata to the defined file path
    
    # return the file path instead of the dictionary of data and metadata
```

Reply:
Thanks for laying this out Carly! I can write the generalized function after a good name. Does "get_forecast_watershed_IO" work or should I specify "ensemble_deterministic" in there and have a longer name?

Reply:
When we demonstrate this one works, we would dissolve the two separate ones, correct?

Reply:
The general "get_watershed_forecast" function is now written and tested to ensure it can handle retrieving either the deterministic or ensemble watershed forecasts in a loop to copy a month's worth of CSV files. This works for a specified or None path.

Resolved in 0b9456fa70d8e4af7558851995470b2606694dcd.